<launch>

    <arg name="video_device" default="/dev/video0" />
    <arg name="camera_frame_id" default="camera_link" />
    <arg name="camera_info" default="camera_info" />
    <arg name="camera_image_raw" default="image_raw" />
    <arg name="camera_model" default="logitech_c150" />
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" clear_params="true" output="screen" >
        <param name="camera_info_url" type="string" value="package://pinteraction/cfg/$(arg camera_model).yaml" />
        <remap from="usb_cam/image_raw" to="$(arg camera_image_raw)" />
        <remap from="usb_cam/camera_info" to="$(arg camera_info)" />
        <param name="video_device" value="$(arg video_device)" />
        <param name="image_width" value="640" />
        <param name="image_height" value="480" />
        <param name="framerate" value="30" />
        <!--param name="contrast" value="128"/>
        <param name="brightness" value="128" />
        <param name="saturation" value="128" /-->
        <param name="pixel_format" value="yuyv" />
        <param name="camera_frame_id" value="$(arg camera_frame_id)" />
        <param name="io_method" value="mmap" />
    </node>

</launch>